src/node.h

```diff
@@ -213,12 +219,10 @@ NODE_EXTERN IsolateData* CreateIsolateData(v8::Isolate* isolate,
                                            struct uv_loop_s* loop);
 NODE_EXTERN void FreeIsolateData(IsolateData* isolate_data);
 
-NODE_EXTERN Environment* CreateEnvironment(IsolateData* isolate_data,
-                                           v8::Local<v8::Context> context,
-                                           int argc,
-                                           const char* const* argv,
-                                           int exec_argc,
-                                           const char* const* exec_argv);
```

Comment:
Do you think you could leave these existing functions, and make them wrappers around the new ones? Otherwise this will have to be considered a semver-major change, which is going to cause trouble for backporting commits that touch the same code…

Reply:
Sounds good. So it's just `env.h`'s `Start()` and `node.h`'s `Init()` that are part of the ABI, right?

What kind of name would you prefer for these functions? To be honest, I can't really think of anything sane... `Start2()` / `Init2()`? `*New()`?

Reply:
AFAICT it's only `node.h` that's public. You can check out the "SDK" https://nodejs.org/dist/v9.0.0/node-v9.0.0-headers.tar.gz

RE name: It's C++ you can just overload...

Reply:
Oh yeah, overloading. Completely forgot about that! I've just added wrappers for the old prototypes. I've marked those `NODE_DEPRECATED`.

Reply:
We consider adding a runtime deprecation as semver-major as well. So the best we can do while still keep this minor, is adding a comment in the code.
Anyway an entry should be added to https://github.com/nodejs/node/blob/master/doc/api/deprecations.md.

Also could you add that (comment & docs) in a separate commit, as I'm not sure everyone will be onboard for deprecating the old way.

@addaleax what do you think?

Reply:
Yeah, I don’t see the need for a runtime deprecation? This is targeted at embedders, and I’d assume they’re used to having to make changes to their code, if not for our APIs then for V8’s.

Reply:
I just added a second commit. Just let me know if there are things I can do to improve the phrasing.